tests/test_diffraction_objects.py

```diff
@@ -341,5 +341,27 @@ def test_dump(tmp_path, mocker):
 
 @pytest.mark.parametrize("inputs, expected", tc_params)
 def test_constructor(inputs, expected):
-    actualdo = DiffractionObject(**inputs)
-    compare_dicts(actualdo.__dict__, expected)
+    actual_do = DiffractionObject(**inputs)
```

Comment:
1) This is to ensure we get `all_arrays` instead of `_all_arrays` and 2) we want to mimic the way the user extracts properties - one by one.